conjure-undertow-annotations/readme.md

```diff
@@ -0,0 +1,184 @@
+# conjure-undertow-processor generated undertow services
+
+`conjure-undertow-processor` generates Undertow handler for services that can not easily be represented using
+[Conjure](https://github.com/palantir/conjure-java).
+
+## Motivation
+
+Historically we have supported Jersey as web server framework, but later migrated to Undertow. Beyond a desire not to
+support multiple frameworks, Jersey is problematic for a variety of other reasons and does not align with our long term
+technical goals of our Java ecosystem.
+
+Consumers that are still using Jersey should migrate and declare their endpoints using Conjure. However, some
+complex services are using features that are tricky to represent or not supported using Conjure. To help these services
+migrate away from Jersey, `conjure-undertow-processor` provides a more flexible way of defining your service interface
+while still profiting from code generation.
+
+## Getting Started
+
+In your API project, add the following dependencies to your `build.gradle` file (make sure you are using 
+[gradle-processors](https://github.com/palantir/gradle-processors)):
+
+```gradle
+dependencies {
+    implementation 'com.palantir.conjure.java:conjure-undertow-annotations'
+    annotationProcessor 'com.palantir.conjure.java:conjure-undertow-processor'
+}
+```
+
+Next, define your service interface and annotate it using [`conjure-undertow-annotations`](src/main/java/com/palantir/conjure/java/undertow/annotations/Handle.java):
+
+```java
+public interface MyService {
+
+    @Handle(method = HttpMethod.POST, path = "/myEndpoint/{pathParam}")
+    MyResponse myEndpoint(
+            AuthHeader authHeader,
+            @Handle.PathParam String pathParam,
+            @Handle.QueryParam(value = "queryParam") Optional<String> queryParam,
```

Comment:
We can simplify many of these annotations (doesn't make much difference, just a tad more concise)
```suggestion
            @Handle.QueryParam("queryParam") Optional<String> queryParam,
```

Reply:
updated!